lib/components/private/InlineField/InlineField.tsx

```diff
@@ -142,19 +191,31 @@ export const InlineField = forwardRef<HTMLElement, InternalInlineFieldProps>(
             type={type}
             id={id}
             name={name}
-            onChange={onChange}
             value={value}
-            checked={checked}
+            onChange={
+              isMixed
+                ? (e: ChangeEvent<HTMLInputElement>) => {
+                    if (ref && typeof ref === 'object' && ref.current) {
+                      ref.current.indeterminate = indeterminateRef.current;
+                    }
+                    if (typeof onChange === 'function') {
+                      onChange(e);
+                    }
+                  }
+                : onChange
+            }
+            checked={checked === 'mixed' ? false : checked}
             position="absolute"
             zIndex={1}
-            className={styles.realField}
+            className={[styles.realField, isMixed ? styles.isMixed : undefined]}
             cursor={!disabled ? 'pointer' : undefined}
             opacity={0}
             aria-describedby={mergeIds(
               ariaDescribedBy,
               message ? messageId : undefined,
               description ? descriptionId : undefined,
             )}
+            aria-checked={isMixed ? 'mixed' : checked}
```

Comment:
Can this just be passed through directly?

Reply:
It prevents javascript consumers from passing "mixed" as the value for `checked` on other implementations of `InlineField`, e.g. `Radio`.